### PR TITLE
[Refactor]: refactor a fraction of test ops

### PR DESCRIPTION
**Pull request by teamwong111**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Refactor test ops and make them more concise. Maybe the PR needs a "WIP" tag.

## Modification

Most of the modification only involves format, little of the modification involves test cases.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by zhouzaida** [comment]
Please fix the CI

**Comment by teamwong111** [comment]
> Please fix the CI

Done

File: tests/test_ops/test_bbox.py [hunk 1]
```diff
@@ -2,41 +2,49 @@
...
+        self.b1_2 = torch.tensor([[1.0, 1.0, 3.0, 4.0], [2.0, 2.0, 3.0,
+                                                         4.0]]).cuda()
+        self.b2_2 = torch.tensor([[0.0, 2.0, 2.0, 5.0], [2.0, 1.0, 3.0,
+                                                         3.0]]).cuda()
+        self.b1_3 = torch.tensor([[0.0, 0.0, 3.0, 3.0]]).cuda()
+        self.b2_3 = torch.tensor([[4.0, 0.0, 5.0, 3.0], [3.0, 0.0, 4.0, 3.0],
+                                  [2.0, 0.0, 3.0, 3.0], [1.0, 0.0, 2.0,
+                                                         3.0]]).cuda()
+
+        self.should_output_1 = np.array([[0.33333334, 0.5], [0.2, 0.5],
```
Comment (zhouzaida) [inline tests/test_ops/test_bbox.py:26]: should_output -> expected_output

File: tests/test_ops/test_bbox.py [hunk 2]
```diff
@@ -2,41 +2,49 @@
...
+from mmcv.ops import bbox_overlaps
+
 
 @pytest.mark.skipif(
     not torch.cuda.is_available(), reason='requires CUDA support')
 class TestBBox(object):
 
-    def _test_bbox_overlaps(self, dtype=torch.float):
+    def setup_class(self):
+        self.b1_1 = torch.tensor([[1.0, 1.0, 3.0, 4.0], [2.0, 2.0, 3.0, 4.0],
```
Comment (zhouzaida) [inline tests/test_ops/test_bbox.py:13]: `b1_1` is not a proper name.
`bbox_1_1` maybe better.

File: tests/test_ops/test_bbox.py [hunk 3]
```diff
@@ -2,41 +2,49 @@
...
+        self.b2_3 = torch.tensor([[4.0, 0.0, 5.0, 3.0], [3.0, 0.0, 4.0, 3.0],
+                                  [2.0, 0.0, 3.0, 3.0], [1.0, 0.0, 2.0,
+                                                         3.0]]).cuda()
+
+        self.should_output_1 = np.array([[0.33333334, 0.5], [0.2, 0.5],
+                                         [0.0, 0.0]])
+        self.should_output_2 = np.array([0.33333334, 0.5])
+        self.should_output_3 = np.array([0, 0.2, 0.5, 0.5])
+
+        self.b1s = [self.b1_1, self.b1_2, self.b1_3]
```
Comment (zhouzaida) [inline tests/test_ops/test_bbox.py:31]: b1s -> bboxes_1

File: tests/test_ops/test_bbox.py [hunk 4]
```diff
@@ -2,41 +2,49 @@
...
+        self.b2s = [self.b2_1, self.b2_2, self.b2_3]
+        self.aligneds = [False, True, False]
```
Comment (zhouzaida) [inline tests/test_ops/test_bbox.py:33]: aligneds -> aligned_flags

File: tests/test_ops/test_bilinear_grid_sample.py [hunk 5]
```diff
@@ -24,17 +25,8 @@ def _test_bilinear_grid_sample(self,
...
-        self._test_bilinear_grid_sample(torch.float, True, -6)
-        self._test_bilinear_grid_sample(torch.float, False, -10)
-        self._test_bilinear_grid_sample(torch.double, True, 5)
-        self._test_bilinear_grid_sample(torch.double, False, 10)
-        self._test_bilinear_grid_sample(torch.double, True, -6)
-        self._test_bilinear_grid_sample(torch.double, False, -10)
+    @pytest.mark.parametrize('dtype', [torch.double, torch.float])
+    @pytest.mark.parametrize('align_corners', [False, True])
+    @pytest.mark.parametrize('multiplier', [1, 10, -5])
+    def test_bilinear_grid_sample(self, dtype, align_corners, multiplier):
```
Comment (zhouzaida) [inline tests/test_ops/test_bilinear_grid_sample.py:31]: It seems like that the unit tests are not consistent with the original.

Comment (teamwong111) [reply tests/test_ops/test_bilinear_grid_sample.py:31]: Yes, but I think the novel tests are also enough to test default, positive and negative numbers. We can add more `multiplier` options, but I think it is unnecessary.

File: tests/test_ops/test_bbox.py (hunk 1 above)
Comment (teamwong111) [reply tests/test_ops/test_bbox.py]: done

File: tests/test_ops/test_bbox.py (hunk 2 above)
Comment (teamwong111) [reply tests/test_ops/test_bbox.py]: done

File: tests/test_ops/test_bbox.py (hunk 3 above)
Comment (teamwong111) [reply tests/test_ops/test_bbox.py]: done

File: tests/test_ops/test_bbox.py (hunk 4 above)
Comment (teamwong111) [reply tests/test_ops/test_bbox.py]: done
